src/converters/table-cell-refresh-post-fixer.js

```diff
@@ -39,3 +39,46 @@ function tableCellRefreshPostFixer( model ) {
 
 	return fixed;
 }
+
+// Check if a table cell in the view requires refreshing.
+//
+// This methods detects changes that will require:
+// - <span> to <p> rename in the view,
+// - adding a missing <paragraph>,
+// - or wrapping a text node in <paragraph>
+//
+// thus requiring refreshing the table cell view.
+//
+// @param {module:engine/model/element~Element} tableCell Table cell to check.
+// @param {String} type Type of change.
+function checkRefresh( tableCell, type ) {
+	// If children of a table cell were removed - refresh it.
+	if ( !tableCell.childCount ) {
+		return true;
+	}
+
+	const children = Array.from( tableCell.getChildren() );
+	const hasInnerText = children.some( child => child.is( 'text' ) );
+
+	// If a bare text node (not wrapped in a paragraph) was added - refresh it.
+	if ( hasInnerText ) {
```

Comment:
Do we have a case when this situation occurs? We fix the table cell with paragraph ASAP, so I am not sure when this could happen 🤔

Reply:
There were tests for this.

Reply:
We concluded that it might be for pasting so let's leave this `if`.